### ISSUE TITLE
Docs build health checks and error resolution

### ISSUE THREAD

**Issue by RichieEscarez** [knative/docs#3278]
Ensure that there are no surprises on a Knative release day and that there are no merged files in the repo that will block the publishing of a new release.

## What is changing? 

Use GitHub Actions to ensure that the knative.dev site build is healthy and block the offending PR that is causing build failures from merging those broken files into the repo.

Currently, all PR automatically get built by Netlify and those build logs (and link to each unique preview) are here:
https://app.netlify.com/sites/knative/deploys

## How will this impact our users?
When peer repos release, the knative/docs repo won't be delayed by hidden/unknown errors.

## In what release will this take happen (to the best of your knowledge)?
v.NEXT

## Context

A file that has been merged into the knative/community or knative/docs repo and has some type of error in it that causes the docs build to fail will block all subsequent builds from publishing.

Repo content only makes it way on to knative.dev on a "successful production" site build and therefore, those files with errors must be fixed to unblock the docs build and allow you to publish a release.

/cc @csantanapr

**Comment by snneji** [comment]
@RichieEscarez does this still apply since we have moved to MkDocs?

**Comment by RichieEscarez** [comment]
> @RichieEscarez does this still apply since we have moved to MkDocs?

Yes I would assume so but I don't know how the mkdocs builds work. @julz and @csantanapr can clarify

**Comment by julz** [comment]
In docs repo I think this is ok now since we have actions that can fail the PR if stuff fails (and PR previews to make sure stuff works). For community repo this is still a problem, though. We've been discussing revisiting the decision to generate (part of) the community docs from the community repo (while understanding why we wanted it done that way originally), which might be one simple way to fix this.
